code/game/machinery/spaceheater.dm

```diff
@@ -234,6 +237,166 @@
 				cell = null
 				. = TRUE
 
+///For use with heating reagents in a ghetto way
+/obj/machinery/space_heater/improvised_chem_heater
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "sheater-off"
+	name = "Improvised chem heater"
+	desc = "A space heater hacked to reroute heating to a water bath on the top."
+	panel_open = TRUE //This is always open - since we've injected wires in the panel
+	//We inherit the cell from the heater prior
+	cell = null
+	///The beaker within the heater
+	var/obj/item/reagent_containers/beaker = null
```

Comment:
QDEL_NULL this on the chem heater's destruction

Reply:
Done